### PR TITLE
[releases/25.x] [Email] Filtering for email retrievals

### PR DESCRIPTION
**Pull request by darjoo**
This pull request backports #2275 to releases/25.x

Fixes [AB#550315](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/550315)

### CODE REVIEW

File: src/System Application/App/Email/src/Account/EmailAccounts.Page.al [hunk 1]
```diff
@@ -457,12 +463,24 @@ page 8887 "Email Accounts"
...
-        V2Filter := Filter;
+        V2V3Filter := Filter;
+    end;
+#endif
+
+    /// <summary>
+    /// Filters the email accounts to only show accounts that are using the Email Connector v2 or v3.
+    /// </summary>
+    /// <param name="Filter">True to filter the email accounts, false to show all email accounts</param>
+    procedure FilterConnectorV3Accounts(Filter: Boolean)
```
Comment (nikolakukrika) [inline src/System Application/App/Email/src/Account/EmailAccounts.Page.al:481]: Minor - UseFilter to indicate it is a boolean?
also filter can clash with built in keywords.

File: src/System Application/App/Email/src/Email/Inbox/EmailRetrievalFilters.Table.al [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    TableType = Temporary;
+    DataClassification = SystemMetadata;
+    InherentEntitlements = RIMDX;
+    InherentPermissions = RIMDX;
+
+    fields
+    {
+        field(1; Id; Integer)
+        {
+            AutoIncrement = true;
```
Comment (nikolakukrika) [inline src/System Application/App/Email/src/Email/Inbox/EmailRetrievalFilters.Table.al:21]: Add captions and tooltips to the fields to the table

File: src/System Application/Test Library/Email/src/TestEmailConnectorv3.Codeunit.al [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.TestLibraries.Email;
+
+using System.Email;
+
+codeunit 134702 "Test Email Connector v3" implements "Email Connector v3"
```
Comment (nikolakukrika) [inline src/System Application/Test Library/Email/src/TestEmailConnectorv3.Codeunit.al:10]: Rename to Mock to indicate that this is a mock

File: src/System Application/Test Library/Email/src/TestEmailConnectorv3.Codeunit.al [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        TestEmailAccount: Record "Test Email Account";
+    begin
+        if TestEmailAccount.Get(AccountId) then
+            exit(TestEmailAccount.Delete());
+        exit(false);
+    end;
+
+    procedure GetLogoAsBase64(): Text
+    begin
+
```
Comment (nikolakukrika) [inline src/System Application/Test Library/Email/src/TestEmailConnectorv3.Codeunit.al:60]: Can we leave the comment why it is blank?

**Review by nikolakukrika — APPROVED** [review]

**Comment by darjoo** [comment]
Closing this backport, will open a new one once the revert goes in as there are some other changes.

**Pull request closed without merging**